nibabel/fileslice.py

```diff
@@ -113,9 +117,16 @@ def canonical_slicers(sliceobj, shape, check_inds=True):
             can_slicers.extend((slice(None),) * n_ellided)
             n_real += n_ellided
             continue
-        # int / slice indexing cases
+        # int / [int] / slice indexing cases
         dim_len = shape[n_real]
         n_real += 1
+
+        # Treat [x] as "x:x+1"
+        ary = np.asanyarray(slicer)
+        if ary.shape == (1,):
+            can_slicers.append(slice(int(ary[0]), int(ary[0]) + 1))
+            continue
```

Comment:
@matthew-brett I made this change to `canonical_slicers` (and a concomitant one to `is_fancy`) because it's a simple enough transform for a common idiom. It allowed me to reuse a lot of the machinery in these to make slicing and affine calculation straightforward.

Hope that's okay.

Reply:
See above - is this just to allow the user to slice the images with single element lists?

Reply:
Yes.

Reply:
Do we need to allow slicing the images with single element lists?  There is some benefit I suppose to sticking to the fancy vs not-fancy distinction - at least it's also what numpy does.